### ISSUE TITLE
Conditional fall back

### ISSUE THREAD

**Issue by dotellie** [windmilleng/tilt#3241]
This request is sort of crazy but would also be quite useful I believe. What I'm thinking is that it would be awesome to have a way to say that the build should fall back to rebuilding the image but only when part of a file has been changed.

I'm currently doing some node.js stuff and the package.json file is... well, quite abused as per usual. Most changes in there don't affect the running application at all, except for the `dependencies` and `devDependencies` sections. Everything else is basically just developer configuration but it still has to rebuild everything just in case I did change any dependencies. If I could tell Tilt either to parse the JSON and check where the differences are or if I could specify some regex for where to look, that would speed up iteration times quite a bit for me.

**Comment by nicks** [comment]
haha, this is the kind of crazy I like though! I feel like there could be some sort of short-circuiting command to detect when changes don't need to trigger further syncs or runs, let me think on this a bit...

Out of curiosity: is source of the slowdown that npm takes a long time to run even if the dependencies/devDependencies haven't changed?

**Comment by dotellie** [comment]
Exactly. Unless I set up some local caching layer for NPM (which is quite a bit more of a hassle than I consider worth) it has to fetch all packages every time from the internet, even if I just changed a script command slightly. This is of course not as much of a problem in the later stages of development when a project stabilizes, but I'm still in the phase where package.json changes are frequent and cause a lot of unnecessary building from Tilt.
